core/src/main/java/io/questdb/cutlass/text/CairoTextWriter.java

```diff
@@ -225,7 +227,8 @@ private void logTypeError(int i) {
 
     private boolean onField(long line, DirectByteCharSequence dbcs, TableWriter.Row w, int i) {
         try {
-            types.getQuick(i).write(w, i, dbcs);
+            final int tableIndex = remapIndex.size() > 0 ? remapIndex.get(i) : i;
```

Comment:
can we create NOOP remap so that we don't have to have conditional here?

Reply:
By NOOP do you mean prefilled i -> i mapping in case no column reorder?

I'm expecting branch predictor optimisation as size() is invariant here.
 Need to test.

Reply:
```
// remapIndex.size() == 0
Benchmark                                     Mode  Cnt   Score   Error  Units
BranchOverIndexAccessBenchmark.testCondition  avgt    5  11.135 ± 0.655  ns/op

// remapIndex.size() == 32
Benchmark                                     Mode  Cnt   Score   Error  Units
BranchOverIndexAccessBenchmark.testCondition  avgt    5  16.546 ± 1.505  ns/op

// remapIndex.size() == 32
Benchmark                 Mode  Cnt   Score   Error  Units
IndexBenchmark.testIndex  avgt    5  16.835 ± 2.407  ns/op

@Benchmark
public long testCondition() {
    long sum = 0;
    for (int i = 0; i < COLUMNS; i++) {
        final int idx = remapIndex.size() > 0 ? remapIndex.get(i) : i;
        sum += data.get(idx);
    }
    return sum;
}

@Benchmark
public long testIndex() {
    long sum = 0;
    for (int i = 0; i < COLUMNS; i++) {
        sum += data.get(remapIndex.get(i));
    }
    return sum;
}
```

Reply:
cool, thank you for the test, i'm convinced :)